### DISCUSSION TITLE
How to use custom tailwind css while building a template?

### DISCUSSION THREAD

**Discussion by GeorgeZard** [fluentcms/FluentCMS#2339] [Q&A] [answered]
I'm pleased to report that the project is now operational again following the previous feedback. Currently, I'm focused on developing a custom template implementation.

## Current Approach
I've created a static HTML template using Tailwind CSS and am systematically breaking it down into generic code blocks. The goal is to provide the community with more flexible, editable code sections.

## Technical Question
I'm seeking guidance on the Tailwind compilation process integration. Specifically, I need to determine whether:

1. I should integrate my Tailwind compilation process with your existing approach, or
2. I can independently compile it within the Template project (i.e., the Fluentcms web app) and proceed from there

## Development Philosophy
My aim is to make this process as fluent as possible (pun intended!) while avoiding potential errors or suboptimal approaches.

I acknowledge that I'm still in the learning phase of understanding the system's complete functionality, and I appreciate your patience as I work through this process.

Thank you again for your continued support, and please maintain this excellent level of work!

**Comment by pournasserian** [comment, marked as answer]
Regarding the template development for editable areas, consider defining `Block` to reuse in website instead of copy/paste the same HTML with different contents.

There is no need to integrate Tailwind compilation process for your own Tailwind classes: we have a custom component for Tailwind which you may insert into your head section in layout file. Here is the usage:
`<TailwindStyles fluentcms />`
 https://github.com/fluentcms/FluentCMS/blob/dev/src/FluentCMS/Templates/Default/AdminLayout.head.html

The component checks if the css is not compiled, it uses client's browser and tailwind.js (interactive mode) to compile the whole page's css and saves {guid}.css file and render the style in page's head (each page has a guid). Any changes in page's content will delete the css and recompile the css.

Of course you always have the option to compile the styles yourself and add it statically to your layout. 

You can use any custom component in your layout by adding `fluentcms` attribute to the tag (like above).

We will be more than happy getting your ideas/feedbacks for process improvement, features, etc.

**Reply by GeorgeZard** [reply]
Thanks for the reply amazing news about tailwind it helps a ton. Yeah i am defining Blocks in the json and implementing different html blocks for alternate html blocks (eg. alternate hero sections). Thanks again !
